pkg/client/edv/client.go

```diff
@@ -32,52 +33,51 @@ func New(edvServerURL string) *Client {
 // CreateDataVault sends the EDV server a request to create a new data vault.
 // The location of the newly created data vault is returned.
 func (c *Client) CreateDataVault(config *operation.DataVaultConfiguration) (string, error) {
-	return c.sendPostJSON(config, "/data-vaults", "a duplicate data vault exists")
+	return c.sendCreateRequest(config, "/data-vaults", "a duplicate data vault exists")
 }
 
 // CreateDocument sends the EDV server a request to store the specified document.
 // The location of the newly created document is returned.
 func (c *Client) CreateDocument(vaultID string, document *operation.StructuredDocument) (string, error) {
-	return c.sendPostJSON(document, fmt.Sprintf("/encrypted-data-vaults/%s/docs", vaultID), "")
+	return c.sendCreateRequest(document, fmt.Sprintf("/encrypted-data-vaults/%s/docs", url.PathEscape(vaultID)), "")
 }
 
-// RetrieveDocument sends the EDV server a request to retrieve the specified document.
+// ReadDocument sends the EDV server a request to retrieve the specified document.
 // The requested document is returned.
-func (c *Client) RetrieveDocument(vaultID, docID string) ([]byte, error) {
-	resp, err := http.Get(c.edvServerURL + "/encrypted-data-vaults/" + vaultID + "/docs/" + docID)
+func (c *Client) ReadDocument(vaultID, docID string) ([]byte, error) {
+	// The linter falsely claims that the body is not being closed
+	// https://github.com/golangci/golangci-lint/issues/637
+	resp, err := http.Get(fmt.Sprintf("%s/encrypted-data-vaults/%s/docs/%s", //nolint: bodyclose
+		c.edvServerURL, url.PathEscape(vaultID), url.PathEscape(docID)))
 	if err != nil {
 		return nil, fmt.Errorf("failed to send GET message: %w", err)
 	}
 
-	defer closeBody(resp)
+	defer closeReadCloser(resp.Body)
 
-	if resp.StatusCode == http.StatusBadRequest {
-		return nil, getError(resp)
-	} else if resp.StatusCode == http.StatusNotFound {
-		return nil,
-			fmt.Errorf(fmt.Sprintf("no document with an id of %s could be found in the vault with id %s", docID, vaultID))
-	}
-
-	return getDocumentFromResponse(resp)
-}
-
-func getDocumentFromResponse(resp *http.Response) ([]byte, error) {
-	respData, err := ioutil.ReadAll(resp.Body)
+	respBytes, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		return nil, fmt.Errorf("failed to read response message while retrieving document: %w", err)
 	}
 
-	return respData, nil
-}
+	if resp.StatusCode == http.StatusOK {
+		return respBytes, nil
+	} else if resp.StatusCode == http.StatusNotFound {
+		serverEndpointReached := string(respBytes) == operation.VaultNotFoundErrMsg ||
```

Comment:
converting bytes to string is probably expensive, convert it once pass it on to some private fuction to make it cleaner/readable.

Reply:
something like 

if resp.StatusCode == http.StatusOK {
}			return respBytes, nil
	} 
else if resp.StatusCode == http.StatusNotFound {
 return checkStatusNotFoundErr()
} else {
//return that generic error
}
`